src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2169,16 +2180,21 @@ internal void GenerateAnonymousFunctionConversionError(BindingDiagnosticBag diag
 
             if (reason == LambdaConversionResult.MismatchedParameterType)
             {
+                // This is checked in the code that returns LambdaConversionResult.MismatchedParameterType
+                Debug.Assert(anonymousFunction.HasExplicitlyTypedParameterList);
+
                 // Cannot convert {0} to type '{1}' because the parameter types do not match the delegate parameter types
+                //
+                // Or: There is a ref-kind mismatch.
```

Comment:
> Or: There is a ref-kind mismatch.

Looks like ref-kind mismatch cannot happen here, it was already checked in the `if` above.

Reply:
indeed!